source/vibe/http/internal/http2.d

```
@@ -1,10 +1,201 @@
+ *   unsupported identifier MUST ignore that setting.
+*/
+alias HTTP2SettingID = ushort;
+alias HTTP2SettingValue = uint;
+
+enum  HTTP2SettingsParameters {
+    headerTableSize                 = 0x1,
+    enablePush                      = 0x2,
+    maxConcurrentStreams            = 0x3,
+    initialWindowSize               = 0x4,
+    maxFrameSize                    = 0x5,
+    maxHeaderListSize               = 0x6
+}
+
+
+struct HTTP2Settings {
+    private:
+        alias H2Params = HTTP2SettingsParameters;
+
+        struct Setting {
+            HTTP2SettingID id;
+            HTTP2SettingValue value;
+        }
+
+        Setting _headerTableSize         = {H2Params.headerTableSize, 4096};
+
+        Setting _enablePush              = {H2Params.enablePush, 1};
+
+        // UNLIMITED, 100 is the minimum recommended value (TODO discuss)
+        Setting _maxConcurrentStreams    = {H2Params.maxConcurrentStreams, 100};
+
+        Setting _initialWindowSize       = {H2Params.initialWindowSize, 65535};
+
+        Setting _maxFrameSize            = {H2Params.maxFrameSize, 16384};
+
+        // UNLIMITED, (TODO discuss);
+        Setting _maxHeaderListSize;
+
+    public:
+        /**
+          * Properties
+          */
+        @property HTTP2SettingValue headerTableSize() @safe { return _headerTableSize.value; }
+
+        @property HTTP2SettingValue enablePush() @safe      { return _enablePush.value; }
+
+        @property HTTP2SettingValue maxConcurrentStreams()
+        @safe {
+            return  _maxConcurrentStreams.value;
+        }
+
+        @property HTTP2SettingValue initialWindowSize()
+        @safe {
+            return _initialWindowSize.value;
+        }
+
+        @property HTTP2SettingValue maxFrameSize() @safe    { return _maxFrameSize.value; }
+
+        @property HTTP2SettingValue maxHeaderListSize()
+        @safe {
+            return _maxHeaderListSize.value;
+        }
+
+        @property void set(ushort code, uint val)
```

This should probably throw/assert if an unsupported code is set. Alternatively, a compile time version could be  made that errors out at compile time (`void set(HTTP2SettingID id)(HTTP2SettingValue value)`). `@property` should be dropped.